odl/solvers/util/steplen.py

```diff
@@ -118,16 +138,20 @@ def __call__(self, x, direction, dir_derivative):
             The current point
         direction : `LinearSpaceElement`
             Search direction in which the line search should be computed
-        dir_derivative : float
+        dir_derivative : float, optional
             Directional derivative along the ``direction``
+            Default: ``function.gradient(x).inner(direction)``
```

Comment:
I don't see an issue with this. In general all of our solvers do not use this default and instead provides this manually since it is easier.